### PR TITLE
feature/api-structure

### PR DESCRIPTION
**Pull request by bchenard**
Setup of the API with an endpoint setup. 

Default port is 5041, get /weatherforecast will give a list of weather forecasts

### CODE REVIEW

**Review by ArthurLeFloch — DISMISSED** [review]
Working as expected.

File: api/api.csproj
Comment (fboitel) [inline api/api.csproj:1]: We will use the same naming convention everywhere : Nebulift.<project>.csproj (all in Pascal Case)

**Review by Zenemis — DISMISSED** [review]
LGTM

File: api/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+[Route("[controller]")]
+public class WeatherForecastController : ControllerBase
+{
+    private static readonly string[] Summaries = new[]
+    {
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+    };
+
+    [HttpGet]
+    public IEnumerable<WeatherForecast> Get()
```
Comment (fboitel) [inline api/Controllers/WeatherForecastController.cs:13]: Every public method should have documentation comment (see https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments)

File: api/api.http
[the remark this answers is not in the document]
Comment (bchenard) [reply api/api.http:1]: added a dedicated folder for .http files

File: api/api.csproj
Comment (bchenard) [reply api/api.csproj:1]: refactoring done

**Review by Zenemis — APPROVED** [review]
LGTM

**Review by Zenemis — DISMISSED** [review]

File: Nebulift.Api/Program.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        app.UseHttpsRedirection();
+        app.UseAuthorization();
+
+        app.MapControllers();
+
+        app.Run();
+    }
+}
+
+public record WeatherForecast(DateOnly Date, int TemperatureC, string? Summary)
```
Comment (theogom) [inline Nebulift.Api/Program.cs:34]: This class should be moved in its own file

**Review by ArthurLeFloch — DISMISSED** [review]

**Comment by theogom** [comment]
We should add namespaces to be able to reference class accross file and project (e.g. `Nebulift`, `Nebulift.Controllers`, `Nebulift.Models`, etc)

File: Nebulift.Api/Program.cs (hunk 2 above)
Comment (bchenard) [reply Nebulift.Api/Program.cs]: moved to models

**Comment by fboitel** [comment]
Folder name should be in accordance with others (infra, frontend)

File: Nebulift.Api/Nebulift.Api.sln
Comment (fboitel) [inline Nebulift.Api/Nebulift.Api.sln:1]: We might want a global solution here that references everything, even if we add other projects in addition to the API (could be a project for models, other jobs, ...). That's why Nebulift.sln fits better here.

**Comment by fboitel** [comment]
Folder Nebulift.Api is no longer needed

File: Nebulift.Api/Nebulift.Api.sln
Comment (bchenard) [reply Nebulift.Api/Nebulift.Api.sln:1]: so change the name of the project  from Nebulift.Api to Nebulift (including namespaces...) ?

File: backend/Nebulift.csproj
Comment (fboitel) [inline backend/Nebulift.csproj:1]: This project is specific to the API, it should be named Nebulift.Api.csproj. In addition it should be in a dedicated folder, see : https://learn.microsoft.com/en-us/dotnet/core/porting/project-structure (In this example, replace car by Api)
Notice that test and source folders are missing as well

File: backend/Program.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        app.UseHttpsRedirection();
+        app.UseAuthorization();
+        app.MapControllers();
+
+        app.Run();
+    }
+}
+
+public record WeatherForecast(DateOnly date, int temperatureC, string? summary)
```
Comment (Zenemis) [inline backend/Program.cs:38]: There now is a WeatherForecast.cs file, meaning the record should be deleted from main program, and the latter file included.

**Review by Zenemis — CHANGES_REQUESTED** [review]

File: backend/Program.cs (hunk 3 above)
Comment (bchenard) [reply backend/Program.cs]: fixed

File: backend/Nebulift.csproj
Comment (bchenard) [reply backend/Nebulift.csproj:1]: done in the new commit

**Review by Zenemis — DISMISSED** [review]

**Review by ArthurLeFloch — APPROVED** [review]

**Review by ArthurLeFloch — APPROVED** [review]

**Review by Zenemis — APPROVED** [review]

**Review by fboitel — APPROVED** [review]

**Pull request merged**